aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java

```diff
@@ -96,7 +166,144 @@ public String getDescription() {
+    } else if (config != null && this.adaptiveSamplingConfig == null) {
+      this.adaptiveSamplingConfig = config;
+
+      // Initialize anomaly capture rate limiter if error capture limit is configured
+      if (config.getErrorCaptureLimit() != null) {
+        int errorSpansPerSecond = config.getErrorCaptureLimit().getErrorSpansPerSecond();
+        this.anomalyCaptureRateLimiter =
+            new RateLimiter(errorSpansPerSecond, errorSpansPerSecond, clock);
+      }
+    }
+  }
+
+  void adaptSampling(ReadableSpan span, SpanData spanData, Consumer<ReadableSpan> spanBatcher) {
+    if (!adaptiveSamplingRuleExists) {
+      return;
+    }
+    Long statusCode = spanData.getAttributes().get(HTTP_RESPONSE_STATUS_CODE);
+
+    boolean shouldBoostSampling = false;
+    boolean shouldCaptureAnomalySpan = false;
+
+    List<AwsXrayAdaptiveSamplingConfig.AnomalyConditions> anomalyConditions =
+        adaptiveSamplingConfig != null ? adaptiveSamplingConfig.getAnomalyConditions() : null;
+    // Empty list -> no conditions will apply and we will not do anything
+    if (anomalyConditions != null && !anomalyConditions.isEmpty()) {
+      String operation = spanData.getAttributes().get(AwsAttributeKeys.AWS_LOCAL_OPERATION);
+      if (operation == null) {
+        operation = generateIngressOperation(spanData);
+      }
+      for (AwsXrayAdaptiveSamplingConfig.AnomalyConditions condition : anomalyConditions) {
+        // Check if the operation matches any in the list or if operations list is null (match all)
+        List<String> operations = condition.getOperations();
+        if (!(operations == null || operations.isEmpty() || operations.contains(operation))) {
+          continue;
+        }
+        // Check if any anomalyConditions detect an anomaly either through error code or latency
+        boolean isAnomaly = false;
+
+        String errorCodeRegex = condition.getErrorCodeRegex();
+        if (statusCode != null && errorCodeRegex != null) {
+          isAnomaly = statusCode.toString().matches(errorCodeRegex);
+        }
+
+        Long highLatencyMs = condition.getHighLatencyMs();
+        if (highLatencyMs != null) {
+          isAnomaly =
+              (errorCodeRegex == null || isAnomaly)
+                  && (span.getLatencyNanos() / 1_000_000.0) >= highLatencyMs;
+        }
+
+        if (isAnomaly) {
+          AwsXrayAdaptiveSamplingConfig.UsageType usage = condition.getUsage();
+          if (usage != null) {
+            switch (usage) {
+              case BOTH:
+                shouldBoostSampling = true;
+                shouldCaptureAnomalySpan = true;
+                break;
+              case SAMPLING_BOOST:
+                shouldBoostSampling = true;
+                break;
+              case ERROR_SPAN_CAPTURE:
+                shouldCaptureAnomalySpan = true;
+                break;
+            }
+          }
+        }
+        if (shouldBoostSampling && shouldCaptureAnomalySpan) {
```

Comment:
For performance, maybe if shouldBoostSampling or shouldCaptureAnomalySpan is true we can skip the remaining BoostSampling or CaptureAnomaly check.